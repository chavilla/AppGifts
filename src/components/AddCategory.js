import React, { useState } from 'react';

import PropTypes from 'prop-types';

const AddCategory = ({setCategory}) => {

    const [inputValue,setInputValue]=useState('');

    const handleChange=(value)=>{
        //console.log('handleChange called');
        setInputValue(value);
    }


    const handleSubmit=(e)=>{
        e.preventDefault();

       // console.log('handle submit');

        if(inputValue.trim().length<3) return false;

        setCategory(state=>[inputValue]);
        setInputValue('');
    }

    return ( 
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type='text'
                onChange={(e)=>handleChange(e.target.value)}
                value={inputValue}
                placeholder='Agrega un HÃ©roe'
            />
        </form>
     );
}

AddCategory.propTypes={
    setCategory:PropTypes.func.isRequired
}

export default AddCategory;