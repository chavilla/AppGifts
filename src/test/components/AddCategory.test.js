import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('addCategoy component', () => {
    
    // Simulate a funtion
    const setCategory=jest.fn();

    let wrapper=shallow(<AddCategory setCategory={setCategory}/>);

    beforeEach(()=>{
        //cuando una funcion es llamada y se quiere limpiar por decirlo asi
        //jest.clearAllMocks();
        wrapper=shallow(<AddCategory setCategory={setCategory}/>);
    })
    
    test('should show snapshot',()=>{    
        expect(wrapper).toMatchSnapshot();
    });

    test('should to change the inputText',()=>{
        
        const input=wrapper.find('input');
        const value= 'Hola Mundo';

        //change event simulator
        input.simulate('change',{target:{ value:value}});

        expect(wrapper.find('p').text().trim()).toBe(value)

    });

    test('should not post any information',()=>{
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategory).not.toHaveBeenCalled();

    });

    test('should to call setCategory and clean the input',()=>{

        const value= 'Hola Mundo';
        //simulate inputchange
        const input=wrapper.find('input');
        input.simulate('change', { target:{ value:value}});
        //simulate inputsubmit
        wrapper.find('form').simulate('submit', { preventDefault(){}});
        // called setCategory at least 1 time
        expect(setCategory).toHaveBeenCalled();
        // value should be empty ''
        expect(input.props().value).toBe('');
    })
        
})