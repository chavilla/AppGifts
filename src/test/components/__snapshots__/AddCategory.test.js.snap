// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addCategoy component should show snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Agrega un HÃ©roe"
    type="text"
    value=""
  />
</form>
`;
